packages/@aws-cdk/aws-cloudfront/README.md

```diff
@@ -140,6 +140,66 @@ new cloudfront.Distribution(this, 'myDist', {
 });
 ```
 
+### Lambda@Edge
+
+Lambda@Edge is an extension of AWS Lambda, a compute service that lets you execute functions that customize the content that CloudFront delivers.
+You can author Node.js or Python functions in the US East (N. Virginia) region,
+and then execute them in AWS locations globally that are closer to the viewer,
+without provisioning or managing servers.
+Lambda@Edge functions are associated with a specific behavior and event type.
+Lambda@Edge can be used rewrite URLs,
+alter responses based on headers or cookies,
+or authorize requests based on headers or authorization tokens.
+
+The following shows a Lambda@Edge function added to the default behavior and triggered on every request:
+
+```typescript
+const myFunc = new lambda.Function(...);
+new cloudfront.Distribution(this, 'myDist', {
+  defaultBehavior: {
+    origin: cloudfront.Origin.fromBucket(myBucket),
+    edgeLambdas: [
+      {
+        lambdaFunction: myFunc.currentVersion,
+        eventType: cloudfront.EventType.VIEWER_REQUEST,
+      }
+    ],
+  },
+});
+```
+
+Lambda@Edge functions can also be associated with additional behaviors,
+either at Distribution creation time,
+or after.
+
+```typescript
+// assigning at Distribution creation
+new cloudfront.Distribution(this, 'myDist', {
+  defaultBehavior: { origin: cloudfront.Origin.fromBucket(myBucket) },
+  additionalBehaviors: {
+    'images/*': {
+      origin: myOrigin,
+      edgeLambdas: [
+        {
+          lambdaFunction: myFunc.latestVersion,
```

Comment:
`!` This can't be `latestVersion`. One of the annoyances/limitations is that the user _must_ specify a numbered version, not $LATEST. See https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/lambda-requirements-limits.html.

We probably want a runtime check at the least to detect when customers have done this and give them a warning pre-deploy.

Reply:
Added check.